src/lib.rs

```diff
@@ -49,22 +49,52 @@ impl<Pat> Grammar<Pat> {
             fn fold_repeat_many(
                 &mut self,
                 elem: RuleWithNamedFields<Pat>,
-                sep: Option<RuleWithNamedFields<Pat>>,
+                sep: Option<(RuleWithNamedFields<Pat>, SepKind)>,
             ) -> RuleWithNamedFields<Pat> {
-                elem.fold(self).repeat_many(Some(
-                    sep.map_or_else(empty, |sep| self.whitespace.clone() + sep.fold(self))
```

Comment:
Wait, this PR is subtly wrong in that it removes `sep.fold` calls. I guess I'll just add them back in on the branch where I noticed they're missing.

Reply:
:( sorry